### PR TITLE
Add paper extension

### PR DESCRIPTION
**Pull request by luctst**
## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

**Comment by pernielsentikaer** [comment]
Did you see [SnapJot](https://www.raycast.com/k41531/snap-jot) before creating this one, this is kinda in the same alley 🙂

**Comment by luctst** [comment]
Hello, I didn't know about this one.

**Comment by pernielsentikaer** [comment]
Could that work for you or should I check for other alternatives, I think we have a few like this extension in the store already 🙂

**Comment by luctst** [comment]
I really appreciate how this extension lets me categorize my ideas, something I haven't found in other extensions. While I'm unable to contribute to other projects right now, I'd love to hear your thoughts. If it's not the right time to publish the extension, I completely understand. Just let me know if we can find alternatives, Thank you! 🙂

File: extensions/paper/src/views/CreatePaper.tsx [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+        <ActionPanel>
+          <Action.SubmitForm title="Create Paper" onSubmit={onSubmit} />
+        </ActionPanel>
+      }
+    >
+      <Form.TextField
+        id="name"
+        value={name}
+        onChange={setName}
+        title="Name of your paper"
```
Comment (pernielsentikaer) [inline extensions/paper/src/views/CreatePaper.tsx:94]:
```suggestion
        title="Name of Your Paper"
```

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi 👋 

Thanks for your contribution 💪 

I have now tested your extension, and I have some feedback ready for you:

- It's possible to create a new category without name, this should be validated. You can use [useForm](https://developers.raycast.com/utilities/react-hooks/useform) for that.

- You can add/edit Paper and leave a blank name, that might not be intended

- Add a shortcut for `Create Paper`

- Instead of pushing you might want `pop()` instead 🙂

I'm looking forward to testing this extension again 🔥 

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

**Comment by luctst** [comment]
Hey 🙂

I've made the changes based on your last review. Just a small thing bugging me about the `pop()` API. So, the deal with `pop()` is that it takes you back to the original command without remounting the component. I need it because I'm using the `useGetConfig` hook in the components' `useEffect`. That's why I was rolling with the `push()` API, 'cause it adds the component to the history stack and remounts it. I know it's not the slickest way to handle this. In the next version, I'm thinking of creating commands instead of views, which seems like a smarter move. If you've got any better ideas, hit me up.

Appreciate your feedback.

**Comment by pernielsentikaer** [comment]
You can see here why `push` is a bad thing

https://github.com/raycast/extensions/assets/38033107/3422f062-d128-4310-b76b-b070a36fcffb

You can achieve the same thing by having it redraw the list, which is done in many extensions, like [Remember the Date](https://www.raycast.com/pernielsentikaer/remember-the-date)

**Comment by luctst** [comment]
Hi, 🙂
I have made the changes. I am now using the `popToRoot` API when submitting the form to reset the view. There should be no more problems

**Comment by mil3na** [comment]
Hi @luctst ! Thanks for your contribution!

By using your extension, I was missing a "create paper" command. It took me some time to understand that it was "hidden" on the action panel. Do you mind adding a command for it?
Also, it would be nice to add a note on the README about how the notes are stored, for transparency purposes.
Other than that, I believe that the extension works as expected and would be a great addition :)

**Comment by luctst** [comment]
Hi @mil3na,
I created a new "create-paper" command and added a brief note to the README file. It should be good. :)

**Review by pernielsentikaer — APPROVED** [review]
Hi 👋 

Looks good to me, approved 🔥

**Pull request merged**